modules/custom/moj_resources/src/PromotedContentApiClass.php

```diff
@@ -153,20 +153,23 @@ private function promotedTags($prison)
     return $this->promotedTerms($tags, $prison);
   }
 
-  private function promotedTerms($data, $prison)
+  private function promotedTerms($terms, $prison)
   {
     $termIds = [];
 
-    foreach ($data as $id => $item) {
-      $termIds[] = $item->tid;
+    foreach ($terms as $id => $term) {
+      array_push($termIds, $term->tid);
     }
 
-    $terms = $this->term_storage->loadMultiple($termIds);
+    $loadedTerms = $this->term_storage->loadMultiple($termIds);
 
-    $promotedTerms = array_filter($terms, function ($item) use ($prison) {
-      if ($item->field_moj_promoted->value == true && $prison == $item->field_promoted_to_prison->target_id) {
+    $promotedTerms = array_filter($loadedTerms, function ($term) use ($prison) {
+      $isPromotedTerm = $term->field_moj_promoted->value == true;
+      $promotedPrisonForTerm = $term->field_promoted_to_prison->target_id;
+      $termIsPromotedToThisPrison = $prison == $promotedPrison;
+      if ($isPromotedTerm && $termIsPromotedToThisPrison) {
```

Comment:
These can be combined as the logic is all AND, so

`if ($isPromotedTerm && ($termIsPromotedToThisPrison || !$promotedPrisonForTerm)) {`

Reply:
I just hate seeing two `return true;` straight after one another 😁